### PR TITLE
Taslima week 6

### PR DESCRIPTION
**Pull request by Taslima26**
# Your Details

Your Name:Taslima
Your City:Manchester
Your Slack Name:Tas

# Homework Details
Hi Mitch, I know some of the functionality still not working but just want to submit and get your feedback about it. 
The start alarm button you have to press twice to get alarm started.
Module:javascript -2 module 3

Week: week -6 javascript

### CODE REVIEW

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,7 +1,60 @@
...
+    clearInterval(timer);
+    }
+    
+    //set up flashing background
+    function setColor() {
+      var x = document.body;
+      x.style.backgroundColor = x.style.backgroundColor == "blue" ? "red" : "blue";
+    }
+
+    var idVar = setInterval(function()  {
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/1-alarmclock/alarmclock.js:38]: hi Tas, when you click on the "Set alarm" button for the first time then it triggers this:
`setInterval(function() {
  setColor()
}, 1000)`
this is causing the background color to change straight away

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js [hunk 2]
```diff
@@ -1,7 +1,60 @@
-function setAlarm() {}
+// Implement the missing functionality according to the task description.
+// Hint: Remember which function you need to call when it's time to alarm.
+  function setAlarm() {
+  let seconds = document.querySelector("#alarmSet").value;
+  document.getElementById("set").addEventListener("click", setup);
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/1-alarmclock/alarmclock.js:5]:
```suggestion
  document.getElementById("set").addEventListener("click", setup);
```
This code here means you have to click on "Set alarm" twice in order to start the count down

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js [hunk 3]
```diff
@@ -1,7 +1,60 @@
...
+
+  var timer;
+  function setup() {
+    clearInterval(timer);
+    timer = setInterval(decrement, 1000);
+  }
+  
+  function decrement() {
+    seconds--;
+    if (seconds === 0) {
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/1-alarmclock/alarmclock.js:17]: This is good - when you've got 0 seconds you want to play the alarm

File: week-6/Homework/mandatory/2-quotegenerator/index.html [hunk 4]
```diff
@@ -11,7 +11,25 @@
...
-  <body>
+  <body id=backgroundColorBody>
+    <div id=quoteDiv>
+      <p id=quoteParagraph>
+        <!-- "Whether you think you can or you think you can’t, you’re right.", -->
+      </p>
+      <p id=quoteAuthorParagraph>
+         <!-- "Henry Ford", -->
+      </p>
+      <button id=newQuoteButton>NewQuote</Button>
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/2-quotegenerator/index.html:22]:
```suggestion
      <button id=newQuoteButton>NewQuote</Button>
```
The button element is nested inside the `quoteDiv` and as a result is changing its position when the quote text changes. Think about how you could rearrange the html slightly so the button position is fixed on the page.

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js [hunk 5]
```diff
@@ -1,7 +1,28 @@
+
+
+function setUp() {
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/2-quotegenerator/quotes.js:3]: Great job here - the changing quote logic is working well

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js [hunk 6]
```diff
@@ -1,7 +1,28 @@
...
+  var randomNumber = Math.floor(Math.random() * quotes.length);
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/2-quotegenerator/quotes.js]:
```suggestion
  const randomNumber = Math.floor(Math.random() * quotes.length);
```

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js [hunk 7]
```diff
@@ -1,7 +1,28 @@
...
+  document.querySelector("#quoteParagraph").innerHTML = quotes[randomNumber].quote;
+  document.querySelector("#quoteAuthorParagraph").innerHTML = quotes[randomNumber].author;
+
+}
+
+var checkBox = document.getElementById("myCheck");
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/2-quotegenerator/quotes.js]:
```suggestion
const checkBox = document.getElementById("myCheck");
```

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 8]
```diff
@@ -1 +1,69 @@
...
+   
+    displayCounter.textContent = `This is an  image number ${count}`;
+    if (count >= imagesArray.length - 1) {
+        count = -1;
+    }
+    
+    count++;
+    return setImage();
+}
+function autoForward() {
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js:40]: Think you need to consider using `setInterval` here in order to make sure the carousel automatically changes

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 9]
```diff
@@ -1 +1,69 @@
+
+
+
 // Write your code here
+//somehow get images in to that div
+//I can load first image initially with 
+//browser start and next button will load remaining images eventually as button pressed ?
+
+
+let imagesArray = ['image1.jpg', 'image2.jpg', 'image3.jpg', 'image4.jpg', 'image5.jpg', 'image6.jpg', 'image7.jpg'];
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js]:
```suggestion
const imagesArray = ['image1.jpg', 'image2.jpg', 'image3.jpg', 'image4.jpg', 'image5.jpg', 'image6.jpg', 'image7.jpg'];
```
I'd consider declaring this variable with the `const` keyword as it doesn't need to be changed over time.

Comment (Dedekind561) [reply week-6/Homework/mandatory/3-slideshow/slideshow.js]: Its contents don't need to be changed over time that is

File: week-6/Homework/mandatory/3-slideshow/slideshow.js [hunk 10]
```diff
@@ -1 +1,69 @@
...
+//this is the div where I will display the images
+let sliderImage = document.querySelector('.sliderImage')
+let count = 0;
+
+function previousImage() {
+    
+    displayCounter.textContent = `This is an  image number ${count}`;
+    if (count <= 0) {
+    //if count passed the full array set the count equal to the length of the array.
```
Comment (Dedekind561) [inline week-6/Homework/mandatory/3-slideshow/slideshow.js:19]: I love the fact that you've written comments on here in order to clarify the steps you've taken in your code 😄

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js (hunk 2 above)
Comment (Taslima26) [reply week-6/Homework/mandatory/1-alarmclock/alarmclock.js:5]: ohh what a silly mistake! Thank you mitch.

File: week-6/Homework/mandatory/1-alarmclock/alarmclock.js (hunk 3 above)
Comment (Taslima26) [reply week-6/Homework/mandatory/1-alarmclock/alarmclock.js:17]: Thanks!

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js (hunk 5 above)
Comment (Taslima26) [reply week-6/Homework/mandatory/2-quotegenerator/quotes.js:3]: Great ! Thank you

File: week-6/Homework/mandatory/2-quotegenerator/quotes.js (hunk 6 above)
Comment (Taslima26) [reply week-6/Homework/mandatory/2-quotegenerator/quotes.js]: I try to cut down on var as much as I can .This is the only one I think. Thank you.

File: week-6/Homework/mandatory/3-slideshow/slideshow.js (hunk 10 above)
Comment (Taslima26) [reply week-6/Homework/mandatory/3-slideshow/slideshow.js:19]: Thank you mitch for all your comments..

**Pull request closed without merging**